src/clover/mains_supply/__utils__.py

```diff
@@ -104,6 +105,20 @@ def get_intermittent_supply_status(  # pylint: disable=too-many-locals
                 else:
                     status.append(0)
         times = pd.DataFrame(status)
+
+        timesdaily = times % 24
```

Comment:
## ❓ I don't think this is doing what you think...
What are you trying to do with this bit of code? I'm not able to follow the logic, so you may want more comments in here...

1. `times` is a dataframe with hours from `0` to the end of the simulation period in its index column, and the availability of the grid in the other column by the looks of it. For the grid called `example`, it looks something like this:
  ```csv
  0   0
1   0
2   0
3   0
4   0
5   0
6   0
7   0
8   1
9   1
10  1
11  1
12  1
13  1
14  1
15  1
16  1
17  1
18  0
19  0
20  0
21  0
22  0
23  0
  ```
  so, by using the remainder operator, `%`, all the values will have the same remainder when divided by 24:
  ```python
  >>> set(pd.DataFrame(times % 24 == times)[0].values)
  {True}
  ```
  I.E., this code doesn't seem to be doing much...
2. By having a probability of availability, and then extending the outage time in this way, your availbility probabilities won't be accurate. E.G., let's say the grid is to be unavailable for 12 hours when there's a blackout. If you simply set the 11 hours following a blackout to be zero, and the grid was originally avilable 23 hours of the day (let's say), then it'll now be available for just 12 hours of the day, so the average avilability will be half what you input. I feel this is misleading (assuming this is what you're trying to do?).